fuel/streams.py

```diff
@@ -37,11 +37,19 @@ class AbstractDataStream(object):
     sources : tuple of strings
         The names of the data sources returned by this data stream, as
         given by the dataset.
+    produces_examples : bool
+        Whether this data stream produces examples (as opposed to batches
+        of examples).
 
     """
     def __init__(self, iteration_scheme=None, axis_labels=None):
         self.iteration_scheme = iteration_scheme
         self.axis_labels = axis_labels
+        if self.iteration_scheme:
+            self.produces_examples = iteration_scheme.requests_examples
+        else:
+            # TODO: is this a valid assumption?
```

Comment:
I think we might be able to make that assumption for `DataStream`, actually, if it doesn't have an iteration scheme, but AbstractDataStream should make no assumptions I don't think, since that'll probably be what other kinds of streams inherit from (honestly, the more I think about it the more I think that `iteration_scheme` actually belongs in `DataStream` rather than `AbstractDataStream`, but let's leave that for now) -- @rizar or @bartvm might have comments on that too).